lib/github_service.rb

```diff
@@ -89,6 +89,20 @@ def milestones(fq_name)
       milestones_cache[fq_name] ||= Hash[service.list_milestones(fq_name, :state => :all).map { |m| [m.title, m.number] }]
     end
 
+    # Cache repositories for an hour, and fetch them if a list they doesn't
+    # exist for a given user
+    def org_repos(org = nil, *args)
+      @org_repos_cache         ||= {}
+      @org_repos_cache_timeout ||= {}
+
+      if @org_repos_cache[org].nil? || @org_repos_cache_timeout[org] < Time.zone.now
+        @org_repos_cache_timeout[org] = 1.hour.from_now
+        @org_repos_cache[org]         = service.org_repos(org, *args).map(&:name)
+      end
```

Comment:
Since the bot runs multi-threaded do we need a mutex to ensure there aren't races?

Reply:
Well, we don't do that for any of the other caches here, so not sure for that here...

That said, now that you mention it, the `RunTest#run_tests` method probably needs a `mutex` since it is doing a `git checkout` effectively, and to calls at the same time could be an issue.

Reply:
Okay, now that `git checkout` isn't happening, I think I have covered the bases that I was leaving this comment open for.  Resolving.